Slim/Control/Queries.pm

```diff
@@ -728,12 +736,14 @@ sub albumsQuery {
 		# Bind selected columns in order
 		my $i = 1;
 		for my $col ( @cols ) {
-		# Adjust column names that are sub-queries to be stored using the AS value
-			if ( $col =~ /SELECT/ ) {
-				my ($newcol) = $col =~ /AS (\w+)/;
+			# Adjust column names that are sub-queries to be stored using the AS value.
+			# The actual column name must be stored as the value to the query.
+			if ($col ne '1') {
+				my $newcol = $c->{$col};
```

Comment:
This relies on `$c->{'whatever'}` always being set to `1` unless we've got a subselect, which I think it is at the moment, but this wasn't always the case: the code below I removed at some point in the last year:
https://github.com/LMS-Community/slimserver/blob/9e7877c9b4946502e71190257462b5b84cee1b0d/Slim/Control/Queries.pm#L573-L574

Reply:
The change should also be made in `_getTagDataForTracks`.

Reply:
We can look at this in two ways: 1.) don't do it, as it could lead to unexpected behaviour, or 2.) the hash has been used to transport data before, let's continue to do so! Yes, the condition would need to be tweaked, eg. by checking for a SQL statement.

Reply:
> The change should also be made in `_getTagDataForTracks`.

Is this in response to your other comment below?

Reply:
> Is this in response to your other comment below?

No, just saying that whatever we end up replacing `if ( $col =~ /SELECT/ ) {` with, it should be done in `_getTagDataForTracks` too, for consistency.